CONTRIBUTING.md

```diff
@@ -176,3 +176,15 @@ We have a lot of mechanisms to help expedite towards an accepted PR.  Here are s
 7. *test, test, test*: pretty self explanatory ([OpenSearchTestCase](./test/framework/src/main/java/org/opensearch/test/OpenSearchTestCase.java) for unit tests, [OpenSearchIntegTestCase](./test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java) for integration & cluster tests, [OpenSearchRestTestCase](./test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java) for testing REST endpoint interfaces, and yaml tests with [ClientYamlTestSuiteIT](./rest-api-spec/src/yamlRestTest/java/org/opensearch/test/rest/ClientYamlTestSuiteIT.java) for REST integration tests)
 
 In general, the more guardrails you add to your change, the higher the chance your PR is merged quickly. We can always relax these guard rails in smaller followup PRs. Reverting a GA feature is much more difficult. Check out the [DEVELOPER_GUIDE](./DEVELOPER_GUIDE.md#submitting-changes) for more useful tips.
+
+## Troubleshooting Failing Builds
+
+The OpenSearch testing framework offers many capabilities but exhibits significant complexity (it does lot of randomization internally to cover as many edge cases and variations as possible). Unfortunately, this posses a challenge by making it harder to discover important issues/bugs in straightforward way and may lead to so called flaky tests - the tests which flip randomly from success to failure without any code changes.
```

Comment:
A bit of grammar and spelling, esp. `poses`

```
The OpenSearch testing framework offers many capabilities, but exhibits significant complexity, and does a lot of randomization internally to cover as many edge cases and variations as possible. Unfortunately, this poses a challenge by making it harder to discover important issues/bugs in a straightforward way, and may lead to so called [flaky tests](https://github.com/opensearch-project/OpenSearch/issues?q=is%3Aissue+is%3Aopen+label%3Aflaky-test) - tests that flip randomly from success to failure without any code changes.
```

Reply:
Sure let me read over what was written and correct some of the writing stuff. I just copied directly from the latest version of the previous PR.